### PR TITLE
SOLR-14474: Fix remaining auxilliary class warnings in Solr

### PR DESCRIPTION
**Pull request by ErickErickson**
Version 2, incorporating D. Smiley's comments, thanks!

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/search/facet/FacetParser.java [hunk 1]
```diff
@@ -16,18 +16,22 @@
...
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.Optional;
+
 import org.apache.solr.common.SolrException;
 import org.apache.solr.common.params.SolrParams;
 import org.apache.solr.common.util.NamedList;
 import org.apache.solr.common.util.StrUtils;
+import org.apache.solr.search.FunctionQParser;
```
Comment (dsmiley) [inline solr/core/src/java/org/apache/solr/search/facet/FacetParser.java:28]: Strange; I wonder why you (your IDE) re-ordered FunctionQParser to a different spot that is no longer alphabetic?  I don't care much about this so you can leave it but it's curious.

File: solr/core/src/java/org/apache/solr/search/facet/FacetParser.java [hunk 2]
```diff
@@ -134,9 +138,9 @@ public Object parseFacetOrStat(String key, String type, Object args) throws Synt
     switch (type) {
       case "field":
       case "terms":
-        return new FacetRequest.FacetFieldParser(this, key).parse(args);
+        return new FacetFieldParser(this, key).parse(args);
```
Comment (dsmiley) [inline solr/core/src/java/org/apache/solr/search/facet/FacetParser.java:141]: Nice; seeing this simple difference makes it clear to me we're now organizing it correctly.

File: solr/core/src/java/org/apache/solr/search/facet/FacetRequest.java [hunk 3]
```diff
@@ -302,7 +298,7 @@ public Query createDomainQuery(FacetContext fcontext) throws IOException {
    */
   public static FacetRequest parse(SolrQueryRequest req, Map<String, Object> params) {
     @SuppressWarnings({"rawtypes"})
-    FacetParser parser = new FacetTopParser(req);
+    FacetParser parser = new FacetParser.FacetTopParser(req);
```
Comment (dsmiley) [inline solr/core/src/java/org/apache/solr/search/facet/FacetRequest.java:301]: Seeing this... I think FacetParser ought to have a static method (`newParser`) without exposing FacetTopParser to outside of itself.  Feel free to leave if you think out of scope but IMO now's a great time.

File: solr/core/src/java/org/apache/solr/search/join/GraphEdgeCollector.java [hunk 4]
```diff
@@ -74,8 +74,10 @@ public void setCollectDocs(FixedBitSet target) {
   }
 
   // the number of docs visited
-  public int getNumHits() { return numHits; }
-  
+  public int getNumHits() {
```
Comment (dsmiley) [inline solr/core/src/java/org/apache/solr/search/join/GraphEdgeCollector.java:77]: I admit, I'm a fan of simple one-liners like how it was :-)

File: solr/core/src/java/org/apache/solr/search/join/GraphEdgeCollector.java [hunk 5]
```diff
@@ -115,87 +117,90 @@ public void doSetNextReader(LeafReaderContext context) throws IOException {
...
 
-  /** Build an automaton to represent the frontier query */
-  private Automaton buildAutomaton(BytesRefHash termBytesHash) {
-    // need top pass a sorted set of terms to the autn builder (maybe a better way to avoid this?)
-    final TreeSet<BytesRef> terms = new TreeSet<BytesRef>();
-    for (int i = 0 ; i < termBytesHash.size(); i++) {
-      BytesRef ref = new BytesRef();
-      termBytesHash.get(i, ref);
-      terms.add(ref);
+    /**
```
Comment (dsmiley) [inline solr/core/src/java/org/apache/solr/search/join/GraphEdgeCollector.java:189]: Again, one-liner javadocs are nice & sweet :-)  (I configure IntelliJ to leave these be)

File: solr/core/src/java/org/apache/solr/update/TransactionLog.java [hunk 6]
```diff
@@ -222,7 +222,8 @@ public boolean writePrimitive(Object val) throws IOException {
   }
 
   // for subclasses
-  protected TransactionLog() {}
+  protected TransactionLog() {
```
Comment (dsmiley) [inline solr/core/src/java/org/apache/solr/update/TransactionLog.java:225]: I guess I failed to convince you :-)

File: solr/core/src/java/org/apache/solr/search/facet/FacetParser.java (hunk 1 above)
Comment (ErickErickson) [reply solr/core/src/java/org/apache/solr/search/facet/FacetParser.java:28]: Haven't the faintest clue whether it was the IDE or me reconciling the imports....

**Pull request closed without merging**